cli/src/semgrep/formatter/text.py

```diff
@@ -491,10 +491,18 @@ def format(
     ) -> str:
         reachable = []
         unreachable = []
-        first_party = []
+        first_party_blocking = []
+        first_party_nonblocking = []
+        first_party_blocking_rules = []
         for match in rule_matches:
             if "sca_info" not in match.extra:
-                first_party.append(match)
+                if match.is_blocking:
+                    first_party_blocking.append(match)
+                    rule_id = match.match.rule_id.value
+                    if rule_id not in first_party_blocking_rules and rule_id != "-":
```

Comment:
Done! I had to sort the set because I wanted to keep the order of the rules listed the same as what you'd see in the list of findings but if listing alphabetically works, I can change `first_party_blocking_rules` to just be initialized as a set, and just add elements to it as well.